src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,110 @@ function updateStatsCheckBox() {
+ * @param {DOM} panels - The panels to be expanded/collapsed. not defined if {@param element} is
+ *         collapse panels button.
+ */
+function expandOrCollapsePanels(element, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    var targetPanels = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(STRING_EXPAND);
+
+    if (isElementAnExpandButton) {
+        expandPanels(element, targetPanels);
+        replaceInHtml(element, STRING_EXPAND, STRING_COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
-            }
+        collapsePanels(targetPanels);
+        replaceInHtml(element, STRING_COLLAPSE, STRING_EXPAND);
+    }
+}
+
+/**
+ * Expands all panels.
+ * @param {DOM} element - The element that was clicked to invoke {@code #expandOrCollapsePanels}
+ */
+function expandPanels(element, panels) {
+    // Expand/collapse buttons on AJAX-loaded panels have a collapse-panels-button class
+    var areAjaxLoadedPanels = $(element).is($('#collapse-panels-button'));
+    var TWENTIETH_SECOND_TO_MILLI_SECOND = 50;
+
+    for (var idx = 0; idx < panels.length; idx++) {
+        expandPanel(panels[idx], idx * TWENTIETH_SECOND_TO_MILLI_SECOND, areAjaxLoadedPanels);
+    }
+}
+
+/**
+ * Expands {@code panel}. If the panel data is not loaded, load it by Ajax first before expansion.
+ * @param {int} timeOut - is in milliseconds
+ */
+function expandPanel(panel, timeOut, isAjaxLoadedPanel) {
+    var isPanelAlreadyExpanded = $(panel).hasClass('in');
+    if (isPanelAlreadyExpanded) {
+        return;
+    }
+
+    if (isAjaxLoadedPanel) {
+        var isElementClickedToLoadAjax = checkAndLoadPanelByAjax(panel);
+        if (isElementClickedToLoadAjax) {
+            // When the element has Ajax class, its panel is in collapsed state. Clicking will expand
+            // its panels.
+            return;
+        }
+    }
+    // expand this panel
+    setTimeout(showSingleCollapse, timeOut, panel);
+}
+
+/**
+ * Checks whether the panel data is loaded. If not, load panel data by Ajax.
+ * @return true if Ajax loading happened
+ */
+function checkAndLoadPanelByAjax(panel) {
+    // When the panel's parent element has class ajax_auto or ajax-response-auto(not both), the panel data
+    // has not been loaded yet. We need to load the panel data by Ajax.
+    // ajax_auto class is for loading normal panel data while ajax-response-auto is for loading users
+    // who have not responded.
+    var $ajaxElement = $(panel).parent().children('.ajax_auto');
+    if ($ajaxElement.length === 0) {
+        $ajaxElement = $(panel).parent().children('.ajax-response-auto');
+    }
+
+    var hasAjaxElement = $ajaxElement.length !== 0;
```

Comment:
Yes. If no Ajax class is found, the hasAjaxElement will be false. This happens when clicking collapse panels button for more than once.